tests/back-end/integration/resources/opportunities/code-with-us.test.ts

```diff
@@ -0,0 +1,199 @@
+import { buildCreateUserParams } from "tests/utils/generate/user";
+import {
+  insertUserWithActiveSession,
+  requestWithCookie
+} from "tests/back-end/integration/helpers/user";
+import { connection } from "tests/back-end/setup-server.jest";
+import { app } from "back-end/index";
+import { agent } from "supertest";
+import { omit, pick } from "lodash";
+import { UserType } from "shared/lib/resources/user";
+import {
+  CWUOpportunityEvent,
+  CWUOpportunityStatus,
+  CreateRequestBody
+} from "shared/lib/resources/opportunity/code-with-us";
+import { buildCWUOpportunity } from "tests/utils/generate/opportunities/code-with-us";
+import { getISODateString } from "shared/lib";
+import { clearTestDatabase } from "tests/back-end/integration/helpers";
+import { adt } from "shared/lib/types";
+
+async function setup() {
+  const [testUser, testSession] = await insertUserWithActiveSession(
+    buildCreateUserParams({ type: UserType.Government }),
+    connection
+  );
+  const appAgent = agent(app);
+  return { testUser, testSession, appAgent };
+}
+
+afterEach(async () => {
+  await clearTestDatabase(connection);
+});
+
+test("code-with-us opportunity crud", async () => {
+  const { testUser, testSession, appAgent } = await setup();
+
+  const opportunity = buildCWUOpportunity();
+  const body: CreateRequestBody = {
```

Comment:
I think the scenario test helpers from the previous PR may help simplify the setup for these tests a fair bit.